tests/moveCues.test.js

```diff
@@ -0,0 +1,629 @@
+import { expect } from 'chai'
+import { moveCues } from '../dist/esm/index.js'
+
+/**
+ * npm run test -- tests/moveCues.test.js
+ */
+
+const cueOrder = [
+  { id: 'cue1' }, // 1
+  { id: 'cue2' }, // 2
+  {
+    id: 'group1', // 3
+    children: [
+      { id: 'cue3' }, // 3.1
+      { id: 'cue4' }, // 3.2
+    ],
+  },
+  { id: 'cue5' }, // 4
+  {
+    id: 'group2', // 5
+    children: [
+      { id: 'cue6' }, // 5.1
+      { id: 'cue7' }, // 5.2
+    ],
+  },
+  { id: 'cue8' }, // 6
+]
```

Comment:
Why not use `{ id: 'cue3.1' }` instead of `{ id: 'cue3' }`, then you can avoid the extra brain effort.

Reply:
Facts. Doing it.